### DISCUSSION TITLE
rabbitmq failed to start

### DISCUSSION THREAD

**Discussion by dipualnur** [rabbitmq/rabbitmq-server#9359] [Other]
### Describe the bug

Hello rabbitmq community,
                                                I am currently facing this error code, as I updated my system ( sudo apt update && sudo apt upgrade) , I tried to diagnosis the error, but as I am not from computer science background didn't understand much.

I am pasting below the output, I am using Linux Ubuntu 22.04 LTS and RAM is 16GB with I311thGen<
Hope, these info can be helpful to clariy.

```
sudo rabbitmq-diagnostics -q status

Error: unable to perform an operation on node 'rabbit@mdmonirul'. Please see diagnostics information and suggestions below.

Most common reasons for this are:

 * Target node is unreachable (e.g. due to hostname resolution, TCP connection or firewall issues)
 * CLI tool fails to authenticate with the server (e.g. due to CLI tool's Erlang cookie not matching that of the server)
 * Target node is not running

In addition to the diagnostics info below:

 * See the CLI, clustering and networking guides on https://rabbitmq.com/documentation.html to learn more
 * Consult server logs on node rabbit@mdmonirul
 * If target node is configured to use long node names, don't forget to use --longnames with CLI tools

DIAGNOSTICS
===========

attempted to contact: [rabbit@mdmonirul]

rabbit@mdmonirul:
  * unable to connect to epmd (port 4369) on mdmonirul: timeout (timed out)

Current node details:
 * node name: 'rabbitmqcli-49-rabbit@mdmonirul'
 * effective user's home directory: /var/lib/rabbitmq
 * Erlang cookie hash: 83qGOFOFZUNL0qnZprcJMg==_
```

**Comment by michaelklishin** [comment]
@dipualnur in order to help you we would need a bit of extra information. Can you please share the last 200 or so lines [of the node log file](https://rabbitmq.com/logging.html#log-file-location)?

If you had RabbitMQ 3.11 before and have upgraded it to 3.12, you must [enable all feature flag](https://rabbitmq.com/feature-flags.html#how-to-start-new-node-disabled-feature-flags) (the Debian package itself does not do that at the moment, one day it will).

**Temporarily** (for one successful start) setting [one environment variable](https://rabbitmq.com/configure.html#environment-env-file-unix) should help:

```
RABBITMQ_FEATURE_FLAGS="restart_streams,stream_sac_coordinator_unblock_group,classic_mirrored_queue_version,direct_exchange_routing_v2,feature_flags_v2,listener_records_in_ets,stream_single_active_consumer,tracking_records_in_ets,classic_queue_type_delivery_support,stream_queue,user_limits,maintenance_mode_status,implicit_default_bindings,quorum_queue,virtual_host_metadata,drop_unroutable_metric,empty_basic_get_metric,delete_ra_cluster_mqtt_node,rabbit_mqtt_qos0_queue"
```

Alternatively, if it's a development node and you don't care about its existing data, you can wipe [its data directory](https://rabbitmq.com/relocate.html) and it should boot. Then enable all feature flags using 

``` shell
rabbitmqctl enable_feature_flag all
```

**Comment by lukebakken** [comment]
Here is the error to investigate:

```
unable to connect to epmd (port 4369) on mdmonirul: timeout (timed out)
```

https://www.google.com/search?q=%22unable+to+connect+to+epmd+(port+4369)%22

I am not sure why this error would show up after an upgrade, however. What is the full output of these commands?

```
epmd -names
ss -4nl
netstat -an
```

Please save the output to a file and ATTACH the file to your response. Do **NOT** paste the output because it will be far too large.

**Reply by michaelklishin** [reply]
That's true. It could be a security-related change, they are not unheard of between upgrades. For example, a new set of firewall defaults that blocks `epmd` port access even for local processes.
